sunkit_image/granule.py

```diff
@@ -0,0 +1,331 @@
+"""
+This module contains functions that will segment images for granule detection.
+"""
+import logging
+
+import numpy as np
+import scipy.ndimage as sndi
+import skimage
+from sklearn.cluster import KMeans
+
+import sunpy
+import sunpy.map
+
+__all__ = [
+    "segment",
+    "get_threshold",
+    "trim_intergranules",
+    "mark_faculae",
+    "kmeans_segment",
+    "cross_correlation",
+]
+
+
+def segment(data_map, skimage_method, res, mark_dim_centers=False):

```

Comment:
```suggestion
def segment(smap, skimage_method, resolution, mark_dim_centers=False):
```
This is a personal nitpick but I think the current variable names could do with some changes.
This will need to be done in the function where they are used and so on. 

On a separate note, is there a reason that `skimage_method` has no default?

Reply:
I will change those names. Not sure why we didn't put a defualt for skimage_method. I will make it defualt to 'li'.